airsonic-main/src/main/java/org/airsonic/player/service/TranscodingService.java

```diff
@@ -202,15 +202,32 @@ public Parameters getParameters(MediaFile mediaFile, Player player, Integer maxB
 
         boolean hls = videoTranscodingSettings != null && videoTranscodingSettings.isHls();
         Transcoding transcoding = getTranscoding(mediaFile, player, preferredTargetFormat, hls);
-        if (transcoding != null) {
-            parameters.setTranscoding(transcoding);
+        Integer bitRate = mediaFile.getBitRate();
+
+        if (mediaFile.isVideo()) {
+            if (bitRate == null) {
+                bitRate = 20000;
```

Comment:
Wouldn't it make sense to use `VideoPlayerController.DEFAULT_BIT_RATE` instead of a magic constant?

Reply:
That is not the maxBitRate, that is the bitRate for the media file itself.  Hopefully it is never null, but I didn't want to allow for that possibility or have to keep checking and handling null values, so this just forces it to be a very large value.

Note that the `VideoPlayerController.DEFAULT_BIT_RATE` will be used since it is less than this large value.

Reply:
I guess I should note that the reason I did it this way is that I am planning for future changes that will allow downsampling of video and other types of media files rather than solely mp3 files.

Reply:
If bitRate is null why even try to set a default? bitRate being null would imply that we don't know what the bitRate of the media file is. In that case the existing behavior of never downsampling seems better than guessing whether to transcode or not.

Reply:
I didn't want to preclude the possibility that airsonic may not know the bitrate in some circumstances (perhaps a media type that Airsonic doesn't know how to detect bitrate but ffmpeg can still transcode?), so I just decided to assume it is max bitrate and let the transcoder fail if it needs to rather than prematurely refusing to do what was asked.